### DISCUSSION TITLE
How to retreive a "reserved IP" created on azure ?

### DISCUSSION THREAD

**Discussion by teto** [pulumi/pulumi#14301] [Pulumi Providers] [answered]
Hi,

I can't suceed in retreiving `Reserved IP address (classic)` resource type via the azure-native provided. 
The resource id follows this pattern `/subscriptions/SUBSCRIPTION_ID/resourceGroups/manual/providers/Microsoft.ClassicNetwork/reservedIps/ip-runner-0` 
getPublicIPAddress seemed like the closest match but pulumi fails with:
`/subscriptions/REDACTED_SUBSCRIPTION_ID/resourceGroups/manual/providers/Microsoft.Network/publicIPAddresses/ip-runner-0`
and we can see the paths dont match. 
I've tried to search the pulumi doc for `Microsoft.ClassicNetwork` without any result (also would be nice to have the pulumi doc point at azure documentation for the matching resource).

Is there any way to do this (in any way) or should I shell out to "az" ? 

For the context, we have CI runners on azure that are whitelisted by their IPs so I want to assign them reserved IPs and deploy those runners via pulumi with those reserved IPs in a "manual" resource group

**Comment by praneetloke** [comment, marked as answer]
It looks like Reserved IPs, part of the Classic Network services, has been discontinued. These services are not part of the ARM API and that's why you can't find a function in the `azure-native` provider to get the reserved IP. Check out the [article about the deprecation](https://azure.microsoft.com/en-us/updates/five-azure-classic-networking-services-will-be-retired-on-31-august-2024/) and how you can migrate to a public IP.

**Reply by teto** [reply]
thanks I had a feeling it was something like this. Reading the doc, it's not clear to me how to reserve an IP and use it in a VM such that it survives the VM destruction (solution seems to rely on a NAT gateway) but it's more a question for azure support.

**Reply by praneetloke** [reply]
A public IP resource if created manually and attached to a VM should survive the VM destruction. I don't know if Azure VMs have the option for auto-assigned public IPs but those would be destroyed along with the VM.

Couple of helpful links:

https://learn.microsoft.com/en-us/azure/virtual-network/ip-services/public-ip-upgrade-classic
https://learn.microsoft.com/en-us/azure/virtual-network/ip-services/create-public-ip-portal?tabs=option-1-create-public-ip-standard

**Reply by teto** [reply]
Because we remove/create resource groups "dev" and "prod" as part of the pulumi script, now we have a resource group "manual" in which we've reserved a few IPs that we associate with the dev/prod VM and it works, e.g. we get static IPs across dev/prod resource group destruction

**Reply by praneetloke** [reply]
[Public IPs](https://www.pulumi.com/registry/packages/azure-native/api-docs/network/publicipaddress/) are the replacement for Classic IPs with similar functionality.
